controllers/nutanixmachine_controller_test.go

```diff
@@ -228,3 +236,133 @@ func TestNutanixMachineReconciler(t *testing.T) {
+	scheme := runtime.NewScheme()
+	err := infrav1.AddToScheme(scheme)
+	require.NoError(t, err)
+	err = capiv1.AddToScheme(scheme)
+	require.NoError(t, err)
+
+	cache := mockctlclient.NewMockCache(mockCtrl)
+
+	mgr := mockctlclient.NewMockManager(mockCtrl)
+	mgr.EXPECT().GetCache().Return(cache).AnyTimes()
+	mgr.EXPECT().GetScheme().Return(scheme).AnyTimes()
+	mgr.EXPECT().GetControllerOptions().Return(config.Controller{MaxConcurrentReconciles: 1}).AnyTimes()
+	mgr.EXPECT().GetLogger().Return(log).AnyTimes()
+	mgr.EXPECT().Add(gomock.Any()).Return(nil).AnyTimes()
+
+	restScope := mockmeta.NewMockRESTScope(mockCtrl)
+	restScope.EXPECT().Name().Return(meta.RESTScopeNameNamespace).AnyTimes()
+
+	restMapper := mockmeta.NewMockRESTMapper(mockCtrl)
+	restMapper.EXPECT().RESTMapping(gomock.Any()).Return(&meta.RESTMapping{Scope: restScope}, nil).AnyTimes()
+
+	mockClient := mockctlclient.NewMockClient(mockCtrl)
+	mockClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	mockClient.EXPECT().RESTMapper().Return(restMapper).AnyTimes()
+
+	reconciler := &NutanixMachineReconciler{
+		Client: mockClient,
+		Scheme: scheme,
+		controllerConfig: &ControllerConfig{
+			MaxConcurrentReconciles: 1,
+		},
+	}
+
+	err = reconciler.SetupWithManager(ctx, mgr)
+	assert.NoError(t, err)
+}
+
+func TestNutanixMachineReconciler_SetupWithManager_BuildError(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	ctx := context.Background()
+	log := ctrl.Log.WithName("controller")
+	scheme := runtime.NewScheme()
+	err := infrav1.AddToScheme(scheme)
+	require.NoError(t, err)
+	err = capiv1.AddToScheme(scheme)
+	require.NoError(t, err)
+
+	cache := mockctlclient.NewMockCache(mockCtrl)
+
+	mgr := mockctlclient.NewMockManager(mockCtrl)
+	mgr.EXPECT().GetCache().Return(cache).AnyTimes()
+	mgr.EXPECT().GetScheme().Return(scheme).AnyTimes()
+	mgr.EXPECT().GetControllerOptions().Return(config.Controller{MaxConcurrentReconciles: 1}).AnyTimes()
+	mgr.EXPECT().GetLogger().Return(log).AnyTimes()
+	mgr.EXPECT().Add(gomock.Any()).Return(errors.New("error")).AnyTimes()
+
+	restScope := mockmeta.NewMockRESTScope(mockCtrl)
+	restScope.EXPECT().Name().Return(meta.RESTScopeNameNamespace).AnyTimes()
+
+	restMapper := mockmeta.NewMockRESTMapper(mockCtrl)
+	restMapper.EXPECT().RESTMapping(gomock.Any()).Return(&meta.RESTMapping{Scope: restScope}, nil).AnyTimes()
+
+	mockClient := mockctlclient.NewMockClient(mockCtrl)
+	mockClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	mockClient.EXPECT().RESTMapper().Return(restMapper).AnyTimes()
+
+	reconciler := &NutanixMachineReconciler{
+		Client: mockClient,
+		Scheme: scheme,
+		controllerConfig: &ControllerConfig{
+			MaxConcurrentReconciles: 1,
+		},
+	}
+
+	err = reconciler.SetupWithManager(ctx, mgr)
+	assert.Error(t, err)
+}
+
+func TestNutanixMachineReconciler_SetupWithManager_ClusterToTypedObjectsMapperError(t *testing.T) {
```

Comment:
comments on the purpose of the test would be appreciated.

Reply:
it's a test for the case when ClusterToTypedObjectsMapper call fails inside SetupWithManager. Largely added to appease the codecov check.